cli/npm/resolvers/local.rs

```diff
@@ -91,7 +91,9 @@ impl LocalNpmPackageResolver {
     specifier: &ModuleSpecifier,
   ) -> Result<PathBuf, AnyError> {
     match self.maybe_resolve_folder_for_specifier(specifier) {
-      Some(path) => Ok(path),
+      // Canonicalize the path so it's not pointing to the symlinked directory
+      // in `node_modules` directory of the referrer.
+      Some(path) => Ok(path.canonicalize()?),
```

Comment:
Let's change this to use `self.fs`. I've wondered if there's a way to have a clippy rule for certain files in the cli directory, but probably not.